### PR TITLE
Added Image Cropping Feature

### PR DESCRIPTION
**Pull request by Pl4gue**
Ref: #36 
I have implemented an image cropping feature which can be used to crop images after selection so that the PDF will be created with the cropped images.

### CODE REVIEW

**Comment by Pl4gue** [comment]
I will try to fix the merge conflict as well

**Comment by Swati4star** [comment]
One bug I noticed here:

I add 5 images by selecting 'select images' option
I click on 'crop images'
I crop first image & click on crop
I press back button for rest of the 4 images.
Click on 'create PDF'
The PDF should contain 5 images, but it only has 1 image.



Another problem,
If I choose x images, you are creating x activities for cropping simultaneously? Won't it affect the processing when x is too large?

File: app/src/main/java/swati4star/createpdf/Home.java [hunk 1]
```diff
@@ -144,6 +158,23 @@ void startAddingImages() {
...
     }
 
+    void cropImages() {
+        int imageCount = 1;
+        if (tempUris.size() == 0) {
+            Toast.makeText(activity, R.string.toast_no_images, Toast.LENGTH_SHORT).show();
+            return;
+        }
+        for (String imageUri : tempUris) {
+            CropImage.activity(Uri.fromFile(new File(imageUri)))
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/Home.java]: This might slow down or hang the phone, in case I select 20 images.

File: app/src/main/java/swati4star/createpdf/Home.java [hunk 2]
```diff
@@ -272,36 +304,38 @@ public void onClick(View v) {
...
-                                Toast.makeText(activity, R.string.toast_name_not_blank, Toast.LENGTH_LONG).show();
-                            } else {
-                                filename = input.toString();
-
-                                new CreatingPdf().execute();
-
-                                if (mMorphCounter1 == 0) {
-                                    mMorphCounter1++;
-                                }
+            if (tempUris.size() == 0) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/Home.java]: why nested if on same condition.

File: app/src/main/java/swati4star/createpdf/Home.java [hunk 3]
```diff
@@ -272,36 +304,38 @@ public void onClick(View v) {
...
+                Toast.makeText(activity, R.string.toast_no_images, Toast.LENGTH_LONG).show();
+                return;
+            } else {
+                imagesUri = (ArrayList<String>) tempUris.clone();
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/Home.java]: It can never reach here.

File: app/src/main/java/swati4star/createpdf/Home.java (hunk 2 above)
Comment (Pl4gue) [reply app/src/main/java/swati4star/createpdf/Home.java]: It's not the same condition. I will first check whether the imageUri List is empty and then, in case the tempUri List isn't empty, clone the tempUri List to the imageUri List, and show the "no images selected" toast in case both lists are empty

File: app/src/main/java/swati4star/createpdf/Home.java (hunk 3 above)
Comment (Pl4gue) [reply app/src/main/java/swati4star/createpdf/Home.java]: It does, if the imageUri List is empty but the tempUri List not.

File: app/src/main/java/swati4star/createpdf/Home.java (hunk 1 above)
Comment (Pl4gue) [reply app/src/main/java/swati4star/createpdf/Home.java]: Do you have a better idea for handling this? I could try to fix this in a way that the next activity only gets started if the previous one has finished.

Comment (Pl4gue) [reply app/src/main/java/swati4star/createpdf/Home.java]: Fixed this issue: There will always only be one CropImage activity running as the next activity gets called only after the getting result of the previous one

File: app/src/main/java/swati4star/createpdf/Home.java (hunk 2 above)
Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/Home.java]: Sorry. My bad. I thought these were same variables.

**Comment by Pl4gue** [comment]
Can you please merge this first? So that I can make a new PR in another branch for the "delete all files" feature

**Comment by Swati4star** [comment]
The code looks file. Just one change. The crop button is made exactly similar to other buttons. This doesn't stand it apart. 
Either 
1. add 'Image Enhancement options' heading at the bottom.
   And add 'Crop Images' as one of them.
   I am thinking of adding effects also later on.
2. If you can come up with some other UI enhancement, feel free to share.

**Comment by Pl4gue** [comment]
Alright, I just added an "Image Enhancement options" box at the bottom, you can insert any kind of new buttons or elements inside the bottom Linearlayout as you want.

<hr>

### This is what it looks like:

<div style="text-align:center" align="middle">

</div>

**Comment by Swati4star** [comment]
Great. Can you please squash these commits into a single one?

Check out https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

**Comment by Pl4gue** [comment]
How can I squash them without getting merge conflicts every time?

**Comment by Pl4gue** [comment]
I have now tried squashing them for 2 hours...

**Comment by Swati4star** [comment]
So, the problem I don't get it why were you even getting merge conflicts in between too. It was you only who was working on your branch. You commit history is quite messed up. Can't merge it directly.

Maybe create a new branch or you can follow this link : https://stackoverflow.com/a/5201642/3623128

**Comment by Swati4star** [comment]
The reason I asked you to squash the changes was that if I would have squashed it, it would have shown the commit under my name. That's why I want you to convert it to a single commit.

**Comment by Pl4gue** [comment]
@Swati4star  I do not have any other idea how I can squash the 3 merge commits. I have again tried for hours but ended up with 3 commits everytime due to the two merge conflicts-solving commits

**Comment by Swati4star** [comment]
Either copy these files somewhere, pull project again & create new PR.
It is not possible to merge this now.
The commit name is also not giving any info about the purpose of this commit.

File: circle.yml [hunk 4]
```diff
@@ -19,3 +19,6 @@ test:
     override:
```
Comment (Swati4star) [inline circle.yml]: Why is this file here? I added it in my commit.

File: config/checkstyle.gradle [hunk 5]
```diff
@@ -0,0 +1,25 @@
+apply plugin: 'checkstyle'
```
Comment (Swati4star) [inline config/checkstyle.gradle]: Same thing for these 2 files too.

**Comment by Pl4gue** [comment]
Ok, I will try with a new PR
